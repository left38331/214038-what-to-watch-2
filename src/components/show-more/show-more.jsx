import React from 'react';
import PropTypes from 'prop-types';

const ShowMore = (props) => {
  return props.shouldShowMore && <div className="catalog__more">
    <button className="catalog__button" type="button" onClick={()=>props.onShowMoreBtnClick()}>Show more</button>
  </div>;
};

ShowMore.propTypes = {
  onShowMoreBtnClick: PropTypes.func.isRequired,
  shouldShowMore: PropTypes.bool.isRequired,
};

export {ShowMore};
