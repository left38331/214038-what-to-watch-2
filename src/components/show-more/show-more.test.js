import React from 'react';
import renderer from 'react-test-renderer';
import {BrowserRouter} from 'react-router-dom';

import {ShowMore} from 'components/show-more/show-more';

const props = {
  onShowMoreBtnClick: () => {},
  shouldShowMore: true
};

it(`Render correctly show-more block component`, () => {
  const showMore = renderer.create(<BrowserRouter><ShowMore {...props} /></BrowserRouter>).toJSON();

  expect(showMore).toMatchSnapshot();
});
