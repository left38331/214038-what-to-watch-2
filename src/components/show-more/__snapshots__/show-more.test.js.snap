// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render correctly show-more block component 1`] = `
<div
  className="catalog__more"
>
  <button
    className="catalog__button"
    onClick={[Function]}
    type="button"
  >
    Show more
  </button>
</div>
`;
